framework/src/meshgenerators/FlipSidesetGenerator.C

```diff
@@ -0,0 +1,72 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FlipSidesetGenerator.h"
+
+#include "CastUniquePointer.h"
+
+registerMooseObject("MooseApp", FlipSidesetGenerator);
+
+InputParameters
+FlipSidesetGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+  params.addClassDescription("A Mesh Generator which flips a given sideset");
+  params.addRequiredParam<MeshGeneratorName>("input", "The mesh we want to modify");
+  params.addRequiredParam<BoundaryName>("boundary", "The sideset (boundary) that will be flipped");
+  return params;
+}
+
+FlipSidesetGenerator::FlipSidesetGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _input(getMesh("input")),
+    _sideset_name(getParam<BoundaryName>("boundary"))
+{
+}
+
+std::unique_ptr<MeshBase>
+FlipSidesetGenerator::generate()
+{
+  // get boundary info
+  BoundaryInfo & boundary_info = _input->get_boundary_info();
+  // get id of the input sideset
+  boundary_id_type sideset_id = boundary_info.get_id_by_name(_sideset_name);
+
+  // Throw an error if the sideset doesn't exist
+  if (sideset_id == libMesh::BoundaryInfo::invalid_id)
+    mooseError("sideset doesn't exist in mesh");
+
+  // get sideset map
+  std::multimap<const Elem *, std::pair<unsigned short int, boundary_id_type>> sideset_map =
+      boundary_info.get_sideset_map();
+
+  // old_elem is the original element attached to the sideset before flipping
+  // new_elem is the element attached to the sideset after flipping
+  for (const auto & [old_elem, id_pair] : sideset_map)
+  {
+    boundary_id_type boundary_id = std::get<1>(id_pair);
+    if (boundary_id == sideset_id)
+    {
+      unsigned short int old_side_id = std::get<0>(id_pair);
+      dof_id_type old_elem_id = old_elem->id();
+      const Elem * new_elem = old_elem->neighbor_ptr(old_side_id);
+
+      // Throw an error if the old element doesn't have a neighbor on the old side
+      if (!new_elem)
+        mooseError("elem " + std::to_string(old_elem_id) +
+                   " doesn't have a neighbor through side " + std::to_string(old_side_id) +
+                   " therefore it can't be flipped");
+
+      unsigned int new_side_id = new_elem->which_neighbor_am_i(old_elem);
+      boundary_info.remove_side(old_elem, old_side_id, sideset_id);
+      boundary_info.add_side(new_elem, new_side_id, sideset_id);
```

Comment:
This surely doesn't work reliably, does it?

Despite the cool C++11 ranged for and C++17 structured binding, under the hood this `for` loop is still just marching an iterator from `sideset_map.begin()` until `sideset_map.end()`.  When you `remove_side` here you're invalidating that iterator and invoking Undefined Behavior.  When you `add_side` here you're inserting new elements that may end up (based on `Elem*` location, unpredictably!) further along in the `multimap`, where your iterator will eventually reach them and try to flip them *again*!

I'm activating our Civet recipes to see what happens there in practice, but IMHO if we don't see some failed tests then that'll just mean we need better test coverage.  ;-)

You can deal with iterators manually to fix the `remove_side` problem, but doing that (outside of libMesh boundary_info.C) breaks our already-shaky encapsulation here even more, and you'll need to create a temporary set to handle the `add_side` problem, so I'd suggest creating temporary sets for both.  Fill the sets in this loop, then operate on them in separate later loops.

Reply:
Oh, wow, I totally missed that you were taking a *copy* of the sideset map, not taking it by reference.  That definitely needed a comment other than "get sideset map"!

That's not as efficient as extra loops (you're copying every entry for every sideset, rather than just every entry for the single sideset you're flipping), but it should work.

Reply:
So, do you prefer that I leave it like that or do it the other way by getting the map by reference and doing the flipping part in a separate loop?

Reply:
A separate loop would have been better, but this way is just fine.